File: docs_old/datatools/codesnippet/CSharp/save-data-to-a-database--multiple-tables-_1.cs
            this.Validate();
            this.ordersBindingSource.EndEdit();
            this.customersBindingSource.EndEdit();

            NorthwindDataSet.OrdersDataTable deletedOrders = (NorthwindDataSet.OrdersDataTable)
                northwindDataSet.Orders.GetChanges(DataRowState.Deleted);

            NorthwindDataSet.OrdersDataTable newOrders = (NorthwindDataSet.OrdersDataTable)
                northwindDataSet.Orders.GetChanges(DataRowState.Added);

            NorthwindDataSet.OrdersDataTable modifiedOrders = (NorthwindDataSet.OrdersDataTable)
                northwindDataSet.Orders.GetChanges(DataRowState.Modified);

            try
            {
                // Remove all deleted orders from the Orders table.
                if (deletedOrders != null)
                {
                    ordersTableAdapter.Update(deletedOrders);
                }

                // Update the Customers table.
                customersTableAdapter.Update(northwindDataSet.Customers);

                // Add new orders to the Orders table.
                if (newOrders != null)
                {
                    ordersTableAdapter.Update(newOrders);
                }

                // Update all modified Orders.
                if (modifiedOrders != null)
                {
                    ordersTableAdapter.Update(modifiedOrders);
                }

                northwindDataSet.AcceptChanges();
            }

            catch (System.Exception ex)
            {
                MessageBox.Show("Update failed");
            }

            finally
            {
                if (deletedOrders != null)
                {
                    deletedOrders.Dispose();
                }
                if (newOrders != null)
                {
                    newOrders.Dispose();
                }
                if (modifiedOrders != null)
                {
                    modifiedOrders.Dispose();
                }
            }

File: docs/data-tools/codesnippet/VisualBasic/save-data-to-a-database-multiple-tables_1.vb
        Me.Validate()
        Me.OrdersBindingSource.EndEdit()
        Me.CustomersBindingSource.EndEdit()

        Dim deletedOrders As NorthwindDataSet.OrdersDataTable = CType(
            NorthwindDataSet.Orders.GetChanges(Data.DataRowState.Deleted), NorthwindDataSet.OrdersDataTable)

        Dim newOrders As NorthwindDataSet.OrdersDataTable = CType(
            NorthwindDataSet.Orders.GetChanges(Data.DataRowState.Added), NorthwindDataSet.OrdersDataTable)

        Dim modifiedOrders As NorthwindDataSet.OrdersDataTable = CType(
            NorthwindDataSet.Orders.GetChanges(Data.DataRowState.Modified), NorthwindDataSet.OrdersDataTable)


        Try
            ' Remove all deleted orders from the Orders table.
            If Not deletedOrders Is Nothing Then
                OrdersTableAdapter.Update(deletedOrders)
            End If

            ' Update the Customers table.
            CustomersTableAdapter.Update(NorthwindDataSet.Customers)

            ' Add new orders to the Orders table.
            If Not newOrders Is Nothing Then
                OrdersTableAdapter.Update(newOrders)
            End If

            ' Update all modified Orders.
            If Not modifiedOrders Is Nothing Then
                OrdersTableAdapter.Update(modifiedOrders)
            End If

            NorthwindDataSet.AcceptChanges()

        Catch ex As Exception
            MsgBox("Update failed")

        Finally
            If Not deletedOrders Is Nothing Then
                deletedOrders.Dispose()
            End If

            If Not newOrders Is Nothing Then
                newOrders.Dispose()
            End If

            If Not modifiedOrders Is Nothing Then
                modifiedOrders.Dispose()
            End If
        End Try